storage/innobase/include/rem0rec.h

```diff
@@ -58,6 +60,11 @@ in addition to the data and the offsets */
 #define REC_STATUS_INFIMUM	2
 #define REC_STATUS_SUPREMUM	3
 
+/* REC_FLAG for instant add columns */
+#define REC_FLAG_NONE           0x00
+#define REC_FLAG_INSTANT        0x01
+#define REC_FLAG_NODE_PTR       0x02
+
```

Comment:
I would prefer `enum` to define, something like this:
```
/** record type (@see function_that_uses_this) */
enum rec_flag_t {
	/** no value for an instant ADD COLUMN is stored, or this is not a clustered index */
	REC_FLAG_NONE,
	/** a value is stored for an instantly added column */
	REC_FLAG_INSTANT,
	/** this is a non-leaf page */
	REC_FLAG_NODE_PTR
};
```
Can we simply use `bool` instead of these defines?
I do not think we really need any NODE_PTR flag in the records. (It was a mistake to add that flag to the status bits of ROW_FORMAT!=REDUNDANT, but I did not want to argue Heikki Tuuri’s decision back then.)
We can use `page_is_leaf(page)` instead.